Artifacts/windows-install-windows-updates/artifact.ps1

```diff
@@ -0,0 +1,77 @@
+###################################################################################################
+
+#
+# PowerShell configurations
+#
+
+# NOTE: Because the $ErrorActionPreference is "Stop", this script will stop on first failure.
+#       This is necessary to ensure we capture errors inside the try-catch-finally block.
+$ErrorActionPreference = "Stop"
+
+# Ensure we set the working directory to that of the script.
+Push-Location $PSScriptRoot
+
+###################################################################################################
+
+function Handle-LastError
+{
+    [CmdletBinding()]
+    param(
+    )
+
+    $message = $error[0].Exception.Message
+    if ($message)
+    {
+        Write-Host -Object "ERROR: $message" -ForegroundColor Red
+        Write-Host -Object $error -ForegroundColor Red
+    }
+    
+    # IMPORTANT NOTE: Throwing a terminating error (using $ErrorActionPreference = "Stop") still
+    # returns exit code zero from the PowerShell script when using -File. The workaround is to
+    # NOT use -File when calling this script and leverage the try-catch-finally block and return
+    # a non-zero exit code from the catch block.
+    exit -1
+}
+
+#
+# Handle all errors in this script.
+#
+trap
+{
+    # NOTE: This trap will handle all errors. There should be no need to use a catch below in this
+    #       script, unless you want to ignore a specific error.
+    Handle-LastError
+}
+
+###################################################################################################
+
+#
+# Main execution block.
+#
+
+try
+{
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+
+    $scriptFolder = Split-Path $Script:MyInvocation.MyCommand.Path
+    $localZipFile = Join-Path $scriptFolder 'PSWindowsUpdate.zip'
```

Comment:
Does `PSWindowsUpdate.zip` have to be embedded? Can we not download it from the internet instead? Would we not want to take advantage of any module updates?

Reply:
Yes, we can download it instead, I'll update to do this